syncs/syncs.go

```diff
@@ -164,19 +164,33 @@ type Map[K comparable, V any] struct {
 	m  map[K]V
 }
 
-func (m *Map[K, V]) Load(key K) (value V, ok bool) {
+// Load loads the value for the provided key and whether it was found.
+func (m *Map[K, V]) Load(key K) (value V, loaded bool) {
 	m.mu.RLock()
 	defer m.mu.RUnlock()
-	value, ok = m.m[key]
-	return value, ok
+	value, loaded = m.m[key]
+	return value, loaded
+}
+
+// LoadFunc calls f with the value for the provided key
+// regardless of whether the entry exists or not.
+// The lock is held for the duration of the call to f.
+func (m *Map[K, V]) LoadFunc(key K, f func(value V, loaded bool)) {
```

Comment:
I think this is ok, but what is the use case?

I hope/assume it's not so a user can mutate `value`?

Reply:
Suppose, I have two disjoint maps, where I must not insert into the other map if the first map already has the entry.

Doing a naïve check this like:
```go
if _, ok := m1.Load(k); !ok {
    m2.Store(k, v)
}
```
is racy since it it's possible that `k` is inserted into `m1` after the check, but before the `m2.Store`.

However, with this I can now do:
```go
m1.LoadFunc(k, func(v V, ok bool) {
    if !ok {
        m2.Store(k, v)
    }
})
```
and atomically update `m2` based on the state of `m1`.

Reply:
> I hope/assume it's not so a user can mutate value?

That would only be possible if `V` is a pointer type, but I don't see how that's any different than:
```go
v, ... := m.Load(k)
*v = ... // mutate the memory backed by v
```
If someone really wants per-entry locking, they should just embed a `sync.Mutex` in `V`.

Reply:
> I can think of a good number of misuses.

True, but I don't think it's any more misuses than the existing API.

With `LoadFunc`, I was planning on hacking the `LoadOrInit` method in gross ways:
```go
_, loaded := m1.LoadOrInit(k, func() (v V) {
    m2.Store(k, v) // not loaded, so store it into m2
    return v // store some sentinel value that indicates non-presence
})
if !loaded {
    m1.Delete(k)
}
```

Reply:
Yup, sounds good. These are some good examples. If you're able to distill some into the docs that might be awesome, but if it's hard to fit that's fine too

Reply:
I'll add an example later, but I'll submit this now so I can use it.